### ISSUE TITLE
"min_paths" and "max_paths" settings in http-c2.json seem to not work properly 

### ISSUE THREAD

**Issue by luca994** [BishopFox/sliver#1286]
Hello!
I noticed that the implant's configuration in charge of the HTTP traffic customization doesn't seem to work as expect.
The "min_paths" parameter specified in the file `.sliver/configs/http-c2.json` does not seem to be correctly interpreted when the urls are generated.

If I set the "min_paths" and the "max_paths" parameters to 1 and I set a single value (e.g. "home") to "session_paths", I expect all paths generated by the agent to be in the form: `/home/filename.ext`
Instead, with these settings, by inspecting the traffic, I found some paths without /home in the path.

The issue seems in the file `sliver/implant/sliver/transports/httpclient/httpclient.go` at line 650.
I am not a go programmer, but I tried to fix this issue by modifying the function randomPath:

```
if 0 < len(segments) {
                n := insecureRand.Intn(len(segments)) // How many segments?
                for index := 0; index < n; index++ {
                        seg := segments[insecureRand.Intn(len(segments))]
                        genSegments = append(genSegments, seg)
                }
        }
```
To this:

```
if 0 < len(segments) {
                min, _ := strconv.Atoi("{{.HTTPC2ImplantConfig.MinPaths}}")
                max, _ := strconv.Atoi("{{.HTTPC2ImplantConfig.MaxPaths}}")
                n := insecureRand.Intn(max-min+1) + min // How many segments?
                for index := 0; index < n; index++ {
                        seg := segments[insecureRand.Intn(len(segments))]
                        genSegments = append(genSegments, seg)
                }
        }
```

Basically, I edited the variable n, which did not properly handled the URI's length.
Once recompiled, sliver seems to work as expected and taking into account the minimum and maximum lengths set in the conf file.

Do you agree about the bug and how it was fixed? In that case I can send a pull request
